exec/java-exec/src/main/java/org/apache/drill/exec/planner/logical/DrillMergeProjectRule.java

```diff
@@ -77,4 +84,79 @@ private boolean checkComplexOutput(Project project) {
     }
     return false;
   }
+
+  @Override
+  public void onMatch(RelOptRuleCall call) {
+    final Project topProject = call.rel(0);
```

Comment:
I'm still a little concerned about this change because lines 90 - 115 are copied from ProjectMergeRule, lines 119 - 124 are new, and lines 126 - 142 are logically equivalent except for the added call to simplifyCast. Considering the similarity of the code and the fact that ProjectMergeRule should do this logic correctly, I have a few questions.

Is this a bug in Calcite ProjectMergeRule or in Drill? If it is a bug in Calcite, shouldn't we be contributing the fix there? If it is a Drill specific bug why is the ProjectMergeRule provided by Calcite not sufficient and why does it cause issues for us when a cast is used?

Reply:
Please see https://issues.apache.org/jira/browse/CALCITE-2223. This fix is meant to be in temporarily until the right fix is agreed upon and implemented in Calcite.

Reply:
@gparai I'm okay with the temporary fix since it has been battle tested with users, but could you  create a Jira to for undoing the changes once the fix in Calcite is made. Also please include some details in the jira about what would need to be done, and add a note in the javadoc of DrillMergeProjectRule pointing to CALCITE-2223 and the Drill Jira. I'm guessing that we would be able to extend from ProjectMergeRule again when the fix for CALCITE-2223 is available?

While it's good to have this fixed in Drill in the short term, in the long term it is not sustainable to accumulate custom fixes for Calcite in Drill. Looking at CALCITE-2223 they haven't settled on a solution yet, and there hasn't been any activity there for two months. I think you should propose this solution for CALCITE-2223 and open a PR against Calcite as well.

Reply:
@ilooner thanks for the comment. I have create DRILL-6501 for the same.

Reply:
Done